[BUG] Iceberg integration tests fail with pyspark.errors.exceptions.captured.UnsupportedOperationException: Creating a view is not supported by catalog: spark_catalog

**Describe the bug**
Build: 
- rapids_it-iceberg-s3tables-dev/39
- rapids_it-iceberg-dev/9

Iceberg integration tests failed with 12 test failures in iceberg_view_test.py. All failures occurred when attempting to create Iceberg views using CREATE VIEW SQL commands. Seems the tests only fail with **spark 3.5.6 + scala2.1.3 + iceberg 1.9.2** and passed with 3.5.5 + 2.12 + iceberg 1.6.1.

Should relate to this change https://github.com/NVIDIA/spark-rapids/pull/14042

```
spark_tmp_table_factory = <conftest.TmpTableFactory object at 0x7f0331033340>
reader_type = 'PERFILE', view_sql = 'SELECT * FROM {table_name}'

    @iceberg
    @ignore_order(local=True)
    @pytest.mark.parametrize('reader_type', rapids_reader_types)
    @pytest.mark.parametrize('view_sql', [
        pytest.param("SELECT * FROM {table_name}", id="select_all"),
        pytest.param("SELECT * FROM {table_name} WHERE _c2 > 0", id="filter_condition"),
        pytest.param("SELECT _c0, _c2, _c6 FROM {table_name}", id="projection"),
        pytest.param("SELECT _c7, COUNT(*) as cnt, SUM(_c2) as sum_c2 FROM {table_name} GROUP BY _c7", id="aggregation"),
    ])
    def test_iceberg_view(spark_tmp_table_factory, reader_type, view_sql):
        """Test reading from an Iceberg view."""
    
        table_name = get_full_table_name(spark_tmp_table_factory)
        view_uuid = str(uuid.uuid4()).replace('-', '_')
        view_name = "iceberg_view_" + view_uuid
    
        # Create an Iceberg table with some data in it
        create_iceberg_table(table_name)
    
        # Create an Iceberg view on the table with a view sql
        def setup_iceberg_view(spark):
            spark.sql(f"CREATE VIEW {view_name} AS {view_sql.format(table_name=table_name)}")
>       with_cpu_session(setup_iceberg_view)

../../src/main/python/iceberg/iceberg_view_test.py:49: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../src/main/python/spark_session.py:149: in with_cpu_session
    return with_spark_session(func, conf=copy)
/opt/conda/lib/python3.10/contextlib.py:79: in inner
    return func(*args, **kwds)
../../src/main/python/spark_session.py:133: in with_spark_session
    ret = func(_spark)
../../src/main/python/iceberg/iceberg_view_test.py:48: in setup_iceberg_view
    spark.sql(f"CREATE VIEW {view_name} AS {view_sql.format(table_name=table_name)}")
../../../spark-3.5.6-bin-hadoop3-scala2.13/python/pyspark/sql/session.py:1631: in sql
    return DataFrame(self._jsparkSession.sql(sqlQuery, litArgs), self)
/home/jenkins/agent/workspace/jenkins-rapids_it-iceberg-s3tables-dev-39-cuda12/jars/spark-3.5.6-bin-hadoop3-scala2.13/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py:1322: in __call__
    return_value = get_return_value(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = ('xro11020', <py4j.clientserver.JavaClient object at 0x7f0355341780>, 'o93', 'sql')
kw = {}, converted = UnsupportedOperationException()

    def deco(*a: Any, **kw: Any) -> Any:
        try:
            return f(*a, **kw)
        except Py4JJavaError as e:
            converted = convert_exception(e.java_exception)
            if not isinstance(converted, UnknownException):
                # Hide where the exception came from that shows a non-Pythonic
                # JVM exception message.
>               raise converted from None
E               pyspark.errors.exceptions.captured.UnsupportedOperationException: Creating a view is not supported by catalog: spark_catalog

../../../spark-3.5.6-bin-hadoop3-scala2.13/python/pyspark/errors/exceptions/captured.py:185: UnsupportedOperationException
```

Test configuration:
- Spark version: 3.5.6
- Scala version: 2.13
- All 12 failures in test_iceberg_view with different reader types (PERFILE, MULTITHREADED, COALESCING) and SQL patterns (select_all, filter_condition, projection, aggregation)

Test results: 12 failed, 148 passed, 541 skipped in 1:33:05

--- a/integration_tests/src/main/python/conftest.py
+++ b/integration_tests/src/main/python/conftest.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2020-2025, NVIDIA CORPORATION.
+# Copyright (c) 2020-2026, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -118,6 +118,10 @@ def is_not_utc():
 def is_iceberg_remote_catalog():
     v = os.environ.get('ICEBERG_TEST_REMOTE_CATALOG')
     return v == "1"
+
+def is_iceberg_rest_catalog():
+    v = os.environ.get('ICEBERG_TEST_CATALOG_TYPE')
+    return v == "rest"
 
 # key is time zone, value is recorded boolean value
 _support_info_cache_for_time_zone = {}

--- a/integration_tests/src/main/python/iceberg/iceberg_view_test.py
+++ b/integration_tests/src/main/python/iceberg/iceberg_view_test.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2025, NVIDIA CORPORATION.
+# Copyright (c) 2025-2026, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -16,13 +16,21 @@ import pytest
 import uuid
 
 from asserts import assert_gpu_and_cpu_are_equal_collect
+from conftest import is_iceberg_rest_catalog
 from data_gen import *
-from iceberg import get_full_table_name, rapids_reader_types, create_iceberg_table
+from iceberg import get_full_table_name, rapids_reader_types, create_iceberg_table, iceberg_base_table_cols, iceberg_gens_list
 from marks import iceberg, ignore_order
 from spark_session import with_cpu_session, is_spark_35x
 
-pytestmark = pytest.mark.skipif(not is_spark_35x(),
-                                reason="Current spark-rapids only support spark 3.5.x")
+pytestmark = [
+    pytest.mark.skipif(not is_spark_35x(),
+                       reason="Current spark-rapids only support spark 3.5.x"),
+
+    # Iceberg view tests only supported with REST catalog,
+    # because the view sql is not supported with Hadoop/S3Table catalog for Iceberg 1.9.x
+    pytest.mark.skipif(not is_iceberg_rest_catalog(),
+                       reason="Iceberg view tests only supported with REST catalog")
+]
 
 @iceberg
 @ignore_order(local=True)
@@ -40,8 +48,12 @@ def test_iceberg_view(spark_tmp_table_factory, reader_type, view_sql):
     view_uuid = str(uuid.uuid4()).replace('-', '_')
     view_name = "iceberg_view_" + view_uuid
 
-    # Create an Iceberg table with some data in it
+    # Create an Iceberg table, and insert data into it
     create_iceberg_table(table_name)
+    def insert_data(spark):
+        df = gen_df(spark, list(zip(iceberg_base_table_cols, iceberg_gens_list)))
+        df.writeTo(table_name).append()
+    with_cpu_session(insert_data)
 
     # Create an Iceberg view on the table with a view sql
     def setup_iceberg_view(spark):

--- a/jenkins/spark-tests.sh
+++ b/jenkins/spark-tests.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 #
-# Copyright (c) 2019-2025, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2019-2026, NVIDIA CORPORATION. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -296,6 +296,7 @@ run_iceberg_tests() {
     ICEBERG_REST_JARS="org.apache.iceberg:iceberg-spark-runtime-${ICEBERG_SPARK_VER}_${SCALA_BINARY_VER}:${ICEBERG_VERSION},\
 org.apache.iceberg:iceberg-aws-bundle:${ICEBERG_VERSION}"
         env \
+          ICEBERG_TEST_CATALOG_TYPE="rest" \
           ICEBERG_TEST_REMOTE_CATALOG=1 \
           PYSP_TEST_spark_driver_memory=6G \
           PYSP_TEST_spark_executor_memory=6G \

